src/pspm_transfer_function.m

```diff
@@ -39,9 +39,15 @@
 
 % check input arguments
 if nargin<1
-    errmsg='No data given.'; warning(errmsg);
+    warning('ID:invalid_input','No data given.'); return;
 elseif nargin<2
-    errmsg='No transfer constant given'; warning(errmsg);
+    warning('ID:invalid_input','No transfer constant given.'); return;
+elseif ~isnumeric(c)
+    warning('ID:invalid_input','The parameter ''c'' has to be numeric.'); return;
+elseif ~isnumeric(Rs)
+    warning('ID:invalid_input','The parameter ''Rs'' has to be numeric.'); return;
+elseif ~isnumeric(offset)
+    warning('ID:invalid_input','The parameter ''offset'' has to be numeric.'); return;

```

Comment:
Note that `Rs` and `offset` are optional input parameters. Therefore, we have to ensure that they were actually passed as input before performing numeric validation.

Reply:
Yeah did not think about that